src/polytopes/ngon.jl

```diff
@@ -129,10 +129,10 @@ function Base.in(p::Point{3,T}, t::Triangle{3,T}) where {T}
   (λ₂ ≥ zero(T)) && (λ₃ ≥ zero(T)) && ((λ₂ + λ₃) ≤ one(T))
 end
 
-function normal(t::Triangle{3})
+function normal(t::Triangle{3,T}) where {T}
   a, b, c = t.vertices
   n = (b - a) × (c - a)
-  n / norm(n)
+  n / norm(n) * oneunit(T)
```

Comment:
I am confused with the usage of oneunit. It seems like what we really want is to propagate units correctly and  that oneunit is just a hack to achieve that. Is there any recommendation in the Unitful.jl docs about propagation of units in generic code? I don' t see why our code with generic T should fail with units. If the coordinates are unitful everything should work no?

Reply:
@juliohm In fact, your understanding is correct. As long as we use the generic T correctly and the correct mathematical description, we don't have to worry about the problem of unit propagation. For this line of code, it needs to be fixed because it is mathematically wrong. Although we really like to get a unit vector by dividing by its modulus length, we rarely notice that the modulus length divided by here should be a number without a unit. Otherwise, dividing two quantities of the same dimension will lose the unit.

In this line, `normalize(n)` may seems better.  Unfortunately, the definition of normalize does not take this issue.  `normalize(n) == n / norm(n)` in current. I think this is a bug in upstream. Before it is realized by Base or Unitful, we have to manually add a oneunit(T) here.

Reply:
In the long run, n / norm(n) is more secure than normalize(n), because it will not be affected by how upstream treats normalize units.